bounded_counter/count_min_sketch.py

```diff
@@ -0,0 +1,99 @@
+import os
+import numpy
+import xxhash
+
+
+def hash_key(i, key):
+    return xxhash.xxh64(str(key), seed=i).intdigest()
+
+algorithm_basic = 0
+algorithm_conservative = 1
+algorithm_logcounter = 2
+
+class CountMinSketch(object):
+    def __init__(self, width=16, depth=8, algorithm = None):
+        self.logshift = 0
+        array_type = numpy.uint32
+        self.algorithm = algorithm_basic
+        self.inc_method = self.basic_increment
+        if algorithm and str(algorithm).lower() == 'conservative':
+            self.algorithm = algorithm_conservative
```

Comment:
If this is the case, better use code injection: let the `algorithm` parameter be a function, not a string. More flexible and cleaner.

Reply:
This doesn't support injectable algorithms. The user can merely choose from available implementations. I'm considering switching to several different implementations possibly using inheritance instead of choosing using input parameter.
What do you think?

Reply:
What's popular in libraries is to allow the user to choose from among a few pre-defined algorithms defined as strings (`algorithm="conservative"` etc), plus allow injecting their object/function in the parameter, for flexibility.

It depends how tightly coupled the algorithms are to `CountMinSketch`. If the interface for `algorithm` is simple (just a few methods, or even a single function?),  if `algorithm` is well encapsulated, then injection is the best choice: `def __init__(..., algorithm=algorithm_conservative)`.